docs/extending/creating_custom_dropdowns.md

```diff
@@ -5,83 +5,140 @@ Follow the steps below to learn how to integrate this component to fit it to you
 
-    Do not remove the `select` input, otherwise form submission may not work.
-
-![Dropdown expanded state](img/dropdown_expanded_state.jpg)
-
-## Generate `<select>` input
-
-Next, generate a standard select input with the `ibexa-custom-dropdown__select` CSS class added to the `<select>` element.
-This element should contain at least one additional attribute: `hidden`. 
-If you want to allow users to pick multiple items from a list, add the `multiple` attribute to the same element.
-
-Example:
-
-```html
-    <select class="ibexa-custom-dropdown__select" hidden multiple></select>
+```twig
+{% set source %}
+<select class="ibexa-input">
+    <option value="DE">DE</option>
+    <option value="US">US</option>
+    <option value="NO">NO</option>
+    <option value="PL">PL</option>
+</select>
+{% endset %}
 ```
 
-![Drop-down multiple selection](img/dropdown_multiple_selection.jpg)
-
-## Add attributes
+ `<select>` input shoud have `ibexa-input` class.
+
+Define `choices`:
+
+```twig
+{% set choices = [{
+    value: "DE",
+    label: "DE"
+}, {
+    value: "US",
+    label: "US"
+}, {
+    value: "NO",
+    label: "NO"
+}, {
+    value: "PL",
+    label: "PL"
+}] %}
+```
 
-Next, look at the `data-value` attribute in the code (line 11 and 12) for duplicated options with the CSS class: `ibexa-custom-dropdown__item`.
-It stores a value of an option from a select input.
+If you need a multilevel `choices`, use the following structure:
+
+```twig
+{% set choices = [{
+    value: "DE",
+    label: "DE"
+}, {
+    value: "US",
+    label: "US"
+}, {
+    value: "NO",
+    label: "NO"
+}, {
+    value: "PL",
+    label: "PL",
+    choices: [{
+        value: "PL_S",
+        label: "Silesian"
+    }, {
+        value: "PL_K",
+        label: "Kashubian"
+    }]
+}] %}
+```
 
-You can provide placeholder text for your custom drop-down. To do so:
+To set `preferred_choices`, use the following:
 
-- put a `data-value` attribute with no value `data-value=""`
-- add a `disabled` attribute to the item in the duplicated list of options to make it unclickable.
+```twig
+{% set preferred_choices = [{
+    value: "NO",
+    label: "NO"
+}] %}
+```
 
+For `value`, see the example:
 
-Example:  
- 
-```html
-<li data-value="" class="ibexa-custom-dropdown__item" disabled>Select an option</li>
-<li data-value="1" class="ibexa-custom-dropdown__item">Option 1</li>
+```twig
+{% set value = [{
+    value: "DE",
+    label: "DE"
+}] %}
 ```
 
+You can configure the following attributes:
+
+Name|Values|Definition|
+|---|------|----------|
+|`source`| - |What is currently defined in the `<select>` input header.|
+|`choices`| - |Elements listed in the drop-down.|
+|`preferred_choices`|Elements listed at the top of the list with a separator.|
+|`value`|-|The currently selected element.It is an object with a key `value`. |
+|`multiple`| true</br>false|Boolean. To allow users to select multiple items, add this attribute to the same element.|
```

Comment:
"the same element" is not clear - does it mean adding `multiple` to every item, or something else?